freeipa/src/main/java/com/sequenceiq/freeipa/controller/FreeIpaV1Controller.java

```diff
@@ -332,6 +336,13 @@ public void stop(@ResourceCrn @NotEmpty String environmentCrn) {
         freeIpaStopService.stop(environmentCrn, accountId);
     }
 
+    @Override
+    @CheckPermissionByResourceCrn(action = AuthorizationResourceAction.ROTATE_SALTUSER_PASSWORD_ENVIRONMENT)
+    public void rotateSaltPassword(@ResourceCrn @NotEmpty String environmentCrn) {
```

Comment:
since environment service calling it, this can be `@InternalOnly`
check authz readme for further details: https://github.com/hortonworks/cloudbreak/blob/master/authorization-common/README.md